api/src/templates/setup/setup.html

```diff
@@ -0,0 +1,163 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>{{ title }}</title>
+  <script>
+    const locale = "{{ locale }}";
+    const getListElem = () => document.getElementById('actions');
+    const clearList = () => {
+      const actionsElem = getListElem();
+      while (actionsElem.firstChild) {
+        actionsElem.removeChild(actionsElem.firstChild);
+      }
+    };
+
+    const createProgressListItem = (action) => {
+      const liElem = document.createElement('li');
+      liElem.className = 'step';
+
+      const loader = document.createElement('div');
+      let loaderClass;
+      if (action.completed) {
+        loaderClass = 'done';
+      } else if (action.started) {
+        loaderClass = 'loading';
+      }
+
+      loader.className = `loader ${loaderClass}`;
+      liElem.appendChild(loader);
+
+      const span = document.createElement('span');
+      span.textContent = action.text;
+      liElem.appendChild(span);
+
+      return liElem;
+    };
+
+    const populateList = (progress) => {
+      const actionsElem = getListElem();
+
+      progress.forEach((action) => {
+        if (!action.display) {
+          return;
+        }
+
+        const listElement = createProgressListItem(action);
+        actionsElem.appendChild(listElement);
+      });
+    }
+
+    const getProgress = async () => {
+
+      const result = await fetch('/progress', { headers: { 'accept-language': locale } });
+      if (result.redirected) {
+        window.location.reload();
+        return;
+      }
+
+      if (!result.ok) {
+        console.error('Error when fetching progress', await result.text());
+        return;
+      }
+
+      const progress = await result.json();
+      clearList();
+      populateList(progress);
```

Comment:
Have you thought about redirecting users to the login page once the setup is complete? For non-admin users trying to log in that's what I'd expect.

Reply:
> Have you thought about redirecting users to the login page once the setup is complete?

This already happens. The request to get progress fails and the page reloads, and loads the login page.